### PR TITLE
sandbox: deny signal to other processes

### PR DESCRIPTION
**Pull request by Moisan**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew typecheck` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

### CODE REVIEW

**Review by Rylan12 — APPROVED** [review]

File: Library/Homebrew/test/sandbox_spec.rb [hunk 1]
```diff
@@ -129,4 +129,22 @@ def initialize(*, **)
       end
     end
   end
+
+  describe "#disallow sending signal to other processes" do
+    # we have to spawn a process, otherwise kill doesn't try to send a signal if the process doesn't exist
+    let(:pid) do
+      pid = spawn("sleep 1000")
+      sleep 1 # Ensure the process has started
```
Comment (Rylan12) [inline Library/Homebrew/test/sandbox_spec.rb:137]: Is this going to be an issue with slowing down the tests, or do we not care enough?

Comment (MikeMcQuaid) [reply Library/Homebrew/test/sandbox_spec.rb:137]:
```suggestion
      sleep 0.1 # Ensure the process has started
```

File: Library/Homebrew/test/sandbox_spec.rb [hunk 2]
```diff
@@ -129,4 +129,22 @@ def initialize(*, **)
       end
     end
   end
+
+  describe "#disallow sending signal to other processes" do
```
Comment (MikeMcQuaid) [inline Library/Homebrew/test/sandbox_spec.rb:133]:
```suggestion
  describe "disallow sending signal to other processes" do
```

**Review by MikeMcQuaid — APPROVED** [review]
Looks good, thanks !

**Review by MikeMcQuaid — APPROVED** [review]

**Review by Rylan12 — APPROVED** [review]

**Pull request merged**